### PR TITLE
fsutils -> Typescript

### PR DESCRIPTION
**Pull request by bkendall**
### Description

- Typescript `fsutils`.
- Fix `nyc` to report on coverage 
	 
### Scenarios Tested

- Tests pass
- Deployed a site with hosting, database rules, and functions

### CODE REVIEW

File: src/fsutils.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as fs from "fs";
```
Comment (mbleigh) [inline src/fsutils.ts]: How about

    import {statSync} from "fs";

File: src/listFiles.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import * as glob from "glob";
 
-export function listFiles(cwd: string, ignore: string[]) {
+export function listFiles(cwd: string, ignore: string[] = []) {
```
Comment (mbleigh) [inline src/listFiles.ts]: Add return type `: string[]`

File: src/fsutils.ts (hunk 1 above)
Comment (bkendall) [reply src/fsutils.ts]: that is possible... I go back and forth haha

File: src/listFiles.ts (hunk 2 above)
Comment (bkendall) [reply src/listFiles.ts]: so, for what it's worth, I tried to find a tslint rule that would enforce that, and I can't. However, it does implicitly know it's `string[]`, which is good. I'll keep looking for the rule config to enforce it.

**Pull request merged**